tests/templates/test_subroutines/test_qpe.py

```diff
@@ -33,11 +33,15 @@ def test_expected_tape(self):
             qml.ControlledQubitUnitary(m @ m, control_wires=[1], wires=[0]),
             qml.Hadamard(2),
             qml.ControlledQubitUnitary(m, control_wires=[2], wires=[0]),
-            qml.QFT(wires=[1, 2]).inv()
+            qml.adjoint(qml.QFT(wires=[1, 2]))
 
         assert len(tape2.queue) == len(tape.queue)
-        assert all([op1.name == op2.name for op1, op2 in zip(tape.queue, tape2.queue)])
-        assert all([op1.wires == op2.wires for op1, op2 in zip(tape.queue, tape2.queue)])
+        for op1, op2 in zip(tape.queue[:-1], tape2.queue[:-1]):
```

Comment:
Just curious: why are you checking this in reverse order?

Reply:
Whoops, never mind. I thought the index was `[::-1]` and not `[:-1]`. Makes sense now 😄

Reply:
We aren't using `qml.equal` on the final operation, because it is an `Adjoint` op, and `qml.equal` doesn't work with nested operators yet.

```suggestion
        # qml.equal doesn't work for Adjoint op yet, so we stop before we get to it.
        for op1, op2 in zip(tape.queue[:-1], tape2.queue[:-1]):
```